src/Core/Resolvers/Sql Query Structures/SqlQueryStructure.cs

```diff
@@ -559,6 +577,27 @@ private SqlQueryStructure(
             ColumnLabelToParam = new();
             FilterPredicates = string.Empty;
             OrderByColumns = new();
+            AddCacheControlOptions(httpRequestHeaders);
+        }
+
+        private void AddCacheControlOptions(IHeaderDictionary? httpRequestHeaders)
+        {
+            // Set the cache control based on the request header if it exists.
+            if (httpRequestHeaders is not null && httpRequestHeaders.TryGetValue(CACHE_CONTROL, out Microsoft.Extensions.Primitives.StringValues cacheControlOption))
+            {
+                CacheControlOption = cacheControlOption;
+            }
+
+            if (!string.IsNullOrEmpty(CacheControlOption) &&
+                !string.Equals(CacheControlOption, CACHE_CONTROL_NO_CACHE, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
We can add these options into a `HashSet` of valid options. Then check for existence of the value `CacheControlOption` in the set, if absent - then error out. Instead of comparing with every single valid option in this manner here.